### PR TITLE
Make Breeze installation and reinstallation support both uv and pipx

### PR DESCRIPTION
**Pull request by potiuk**
So far `breeze` fully supported only `pipx` installation. For example it would reinstall itself automatically with pipx if you attempted to run it from another workspace/checked out repository of Airflow, and it only provided instructions for pipx.

With this PR:

* the `uv tool` is preferred way to install breeze
* the `pipx` is PSF-governance managed alternative
* breeze will reinstall itself using uv if it has been installed with uv before when it is run from a different workspace or different airflow repo checked out in another folder

Also documentation is updated to make `uv` the recommended tool and describing how to install it - with `pipx` provided as an alternative.

Warning is printed in case pre-commit-uv is not installed with the
pre-commit (pre-commit-uv significantly speeds up installation of
the venvs by pre-commit). This warning also provides instructions
how to install it.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
We'll have to cherry-pick that one to v2-10-test to enable automated breeze switching between checked out `main` and `v2-10-test` to the right version

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
I also tested a few scenarios where we were switching between "breeze static checks" and "git commit" pre-commit as I noticed that they could be using different pre-commit installation, and they could make no use of the "pre-commit-uv" - I fixed it in this PR, and now also users will get warning if their pre-commit is installed without `uv` and they also will get instructions how to install it.

**Review by jscheffl — APPROVED** [review]

File: dev/breeze/doc/01_installation.rst [hunk 1]
```diff
@@ -151,13 +151,28 @@ Docker in WSL 2
     If VS Code is installed on the Windows host system then in the WSL Linux Distro
     you can run ``code .`` in the root directory of you Airflow repo to launch VS Code.
 
-The pipx tool
---------------
+The uv tool
+-----------
+
+We are recommending to use the ``uv`` tool to manage your virtual environments and generally as a swiss-knife
```
Comment (amoghrajesh) [inline dev/breeze/doc/01_installation.rst:157]: Nice analogy :)

File: dev/breeze/doc/01_installation.rst [hunk 2]
```diff
@@ -151,13 +151,28 @@ Docker in WSL 2
...
+of your Python environment (it supports installing various versions of Python, creating virtual environments,
+installing packages, managing workspaces and running development tools.).
+
+Installing ``uv`` is described in the `uv documentation <https://docs.astral.sh/uv/getting-started/installation/>`_.
+We highly recommend using ``uv`` to manage your Python environments, as it is very comprehensive,
+easy to use, it is faster than any of the other tools availables (wey faster!) and has a lot of features
```
Comment (amoghrajesh) [inline dev/breeze/doc/01_installation.rst]:
```suggestion
easy to use, it is faster than any of the other tools availables (way faster!) and has a lot of features
```

File: dev/breeze/doc/01_installation.rst [hunk 3]
```diff
@@ -151,13 +151,28 @@ Docker in WSL 2
...
+that make it easier to work with Python.
+
+Alternative: pipx tool
+----------------------
 
-We are using ``pipx`` tool to install and manage Breeze. The ``pipx`` tool is created by the creators
+However, we do not want to be entirely dependent on ``uv`` as it is a software governed by e VC-backed vendor,
```
Comment (amoghrajesh) [inline dev/breeze/doc/01_installation.rst]:
```suggestion
However, we do not want to be entirely dependent on ``uv`` as it is a software governed by a VC-backed vendor,
```

File: dev/breeze/doc/01_installation.rst [hunk 4]
```diff
@@ -151,13 +151,28 @@ Docker in WSL 2
...
+so we always want to provide open-source governed alternatives for our tools. If you can't or do not want to
+use ``uv` we got you covered. Another too you can use to manage development tools (and ``breeze`` development
```
Comment (amoghrajesh) [inline dev/breeze/doc/01_installation.rst]:
```suggestion
use ``uv``, we got you covered. Another too you can use to manage development tools (and ``breeze`` development
```

**Review by amoghrajesh — APPROVED** [review]
Few nits, otherwise LGTM +1

File: dev/breeze/doc/01_installation.rst (hunk 1 above)
Comment (potiuk) [reply dev/breeze/doc/01_installation.rst:157]: Indeed :). They are a swiss-knife like :)

**Pull request merged**